src/expr/src/scalar/func/impls/datum.rs

```diff
@@ -21,6 +21,36 @@ sqlfunc!(
     }
 );
 
+sqlfunc!(
+    #[sqlname = "istrue"]
+    #[propagates_nulls = false]
+    #[introduces_nulls = false]
+    #[preserves_uniqueness = false]
+    fn is_true(a: Datum<'_>) -> Result<Option<bool>, EvalError> {
+        Ok(Some(a == Datum::True))
+    }
+);
+
+sqlfunc!(
+    #[sqlname = "isfalse"]
+    #[propagates_nulls = false]
+    #[introduces_nulls = false]
+    #[preserves_uniqueness = false]
+    fn is_false(a: Datum<'_>) -> Result<Option<bool>, EvalError> {
+        Ok(Some(a == Datum::False))
+    }
+);
+
+sqlfunc!(
+    #[sqlname = "isunknown"]
+    #[propagates_nulls = false]
+    #[introduces_nulls = false]
+    #[preserves_uniqueness = false]
+    fn is_unknown(a: Datum<'_>) -> Result<Option<bool>, EvalError> {
+        Ok(Some(a == Datum::Null))
+    }
+);
+
```

Comment:
So actually no need to introduce new UnaryFunc variants in this case! Instead you can plan `IS TRUE` as `BinaryFunc::Eq(<expr>, Datum::True)`, `IS FALSE` as `BinaryFunc::Eq(<expr>, Datum::False)`, and `IS UNKNOWN` as `UnaryFunc::IsNull(<expr>)`.

Reply:
I took a stab at this route and unfortunately `BinaryFunc::Eq` propagates NULLs which is not the behavior we want here.  e.g. `NULL IS TRUE` should return `false` not `NULL`.  It seemed worse to introduce a generic `EqWithoutPropagatingNull` function than to have a couple special functions.  What do you think?

I did go ahead and get rid of the special `IsUnknown`

Reply:
Aw shoot, sorry, I forgot #7597 hasn't been merged. Can you instead drop a TODO that `IsTrue` and `IsFalse` can be replaced with `BinaryFunc::eq` once #7581 is fixed?